doc/src/Developer_write_pair.rst

```diff
@@ -0,0 +1,1196 @@
+Writing new pair styles
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Pair styles are at the core of most simulations with LAMMPS, since they
+are used to compute the forces (plus energy and virial contributions, if
+needed) on atoms for pairs of atoms within a given cutoff.  This is
+often the dominant computation in LAMMPS, and sometimes even the only
+one.  Pair styles can be grouped into multiple categories:
+
+#. simple pairwise additive interactions of point particles
+   (e.g. :doc:`Lennard-Jones <pair_lj>`, :doc:`Morse <pair_morse>`,
+   :doc:`Buckingham <pair_buck>`)
+#. pairwise additive interactions of point particles with added
+   :doc:`Coulomb <pair_coul>` interactions or *only* the Coulomb interactions
+#. manybody interactions of point particles (e.g. :doc:`EAM <pair_eam>`,
+   :doc:`Tersoff <pair_tersoff>`)
+#. complex interactions that include additional per-atom properties
+   (e.g. Discrete Element Models (DEM), Peridynamics, Ellipsoids)
+#. special purpose pair styles that may not even compute forces like
+   :doc:`pair_style zero <pair_zero>` and :doc:`pair_style tracker
+   <pair_tracker>`, or are a wrapper for multiple kinds of interactions
+   like :doc:`pair_style hybrid <pair_hybrid>`, :doc:`pair_style list <pair_list>`,
+   and :doc:`pair_style kim <pair_kim>`
+
+In the text below, we will discuss aspects of implementing pair styles
+in LAMMPS by looking at representative case studies.  The design of
+LAMMPS allows developers to focus on the essentials, which is to compute
+the forces (and energies or virial contributions), enter and manage the
+global settings as well as the potential parameters, and the pair style
+specific parts of reading and writing restart and data files.  Most of
+the complex tasks like management of the atom positions, domain
+decomposition and boundaries, or neighbor list creation are handled
+transparently by other parts of the LAMMPS code.
+
+As shown on the page for :doc:`writing or extending pair styles
+<Modify_pair>`, in order to implement a new pair style, a new class must
+be written that is either directly or indirectly derived from the
+``Pair`` class.  If that class is directly derived from ``Pair``, there
+are three methods that *must* be re-implemented, since they are "pure"
+in the base class: ``Pair::compute()``, ``Pair::settings()``,
+``Pair::coeff()``.  In addition, a custom constructor is needed.  All
+other methods are optional and have default implementations in the base
+class (most of which do nothing), but they may need to be overridden
+depending on the requirements of the model.
+
+Case 1: a pairwise additive model
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this section, we will describe the procedure of adding a simple pair
+style to LAMMPS: an empirical model that can be used to model liquid
+mercury.
```

Comment:
also say which pair style this is within LAMMPS and where the file is, so users can look at the actual pair file as they read this section ?